context/app/static/js/pages/WorkspacePleaseWait/WorkspacePleaseWait.jsx

```diff
@@ -0,0 +1,32 @@
+import React, { useContext, useState } from 'react';
+
+import { AppContext } from 'js/components/Providers';
+
+function WorkspacePleaseWait({ id }) {
+  const { workspacesEndpoint, workspacesToken } = useContext(AppContext);
+
+  const [status, setStatus] = useState();
+
+  async function loadIfUp() {
+    const response = await fetch(`${workspacesEndpoint}/jobs/${id}`, {
+      headers: {
+        'Content-Type': 'application/json',
+        'UWS-Authorization': `Token ${workspacesToken}`,
+      },
+    });
+    const responseJson = await response.json();
+    const { current_job_details } = responseJson.data.jobs[0].job_details;
+    setStatus(current_job_details.message);
+    const { connection_details } = current_job_details;
+    if (connection_details) {
+      const { url_path, url_domain } = connection_details;
+      document.location = `${url_domain}${url_path}`;
+    }
+  }
+
+  setInterval(loadIfUp, 1000);
```

Comment:
I'm unfamiliar with the performance of the workspaces API, but it could be problematic if the request takes longer than the interval to return.

https://developer.mozilla.org/en-US/docs/Web/API/setInterval#ensure_that_execution_duration_is_shorter_than_interval_frequency